src/stamps.ts

```diff
@@ -300,6 +315,30 @@ export class StampsManager {
     this.extendingStamps.splice(stampIndex, 1)
   }
 
+  public async refreshStampsExtendsCapacity(config: StampsConfigExtendsCapacity, beeDebug: BeeDebug): Promise<void> {
+    stampCheckCounter.inc()
+    logger.info('checking postage stamps')
+    const stamps = await beeDebug.getAllPostageBatch()
+    logger.debug('retrieved stamps', stamps)
+
+    const { usageThreshold } = config
+
+    // Get all used stamps sorted by usage from most used to least
+    this.usableStamps = filterUsableStampsExtendsCapacity(stamps, usageThreshold)
+
+    for (const stamp of this.usableStamps) {
+      try {
+        logger.debug(`extending stamp capacity: ${stamp.batchID}`)
+        await topUpStamp(beeDebug, stamp.batchID, (Number(stamp.amount) * 2).toString())
```

Comment:
Oooooh yeah, this is a problem 🙈 The reason why the `amount` is a string and not a number is that we are calculating with very big amounts which are hitting the limits of the JS number engine (see for example [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)). Sooo, unfortunately, you can't just convert it to a number and do the math as there is a potential loss of precision in the game, which generally is a bad idea when you are working with money ;-)

I would suggest using `BigInt` which is native in JS (in Node it is available since 10.x so it is safe to use now as we support >=14.x), please see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) for more details on how to use it.